fix(autoware_behavior_path_start_planner_module): fix shadowVariable

kobayu858 opened the pull request:
## Description
This is a fix based on cppcheck shadowVariable warnings

```
planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:50:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.object_types_to_check_for_path_generation.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:117:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:144:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.search_configs.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:160:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.costmap_configs.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:166:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.astar.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:175:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.rrtstar.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:315:17: style: Local variable 'obj_types_ns' shadows outer variable [shadowVariable]
    std::string obj_types_ns = surround_moving_obstacle_check_ns + "object_types_to_check.";
                ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:459:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:489:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.search_configs.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:508:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.costmap_configs.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:516:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.astar.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:525:23: style: Local variable 'ns' shadows outer variable [shadowVariable]
    const std::string ns = "start_planner.freespace_planner.rrtstar.";
                      ^

planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:698:17: style: Local variable 'obj_types_ns' shadows outer variable [shadowVariable]
    std::string obj_types_ns = surround_moving_obstacle_check_ns + "object_types_to_check.";
                ^
```

## Related links

**Parent Issue:**

- Link

<!-- ⬇️🟢
**Private Links:**

- [CompanyName internal link]()
⬆️🟢 -->

## How was this PR tested?

## Notes for reviewers

None.

## Interface changes

None.

<!-- ⬇️🔴

### Topic changes

#### Additions and removals

| Change type   | Topic Type      | Topic Name    | Message Type        | Description       |
|:--------------|:----------------|:--------------|:--------------------|:------------------|
| Added/Removed | Pub/Sub/Srv/Cli | `/topic_name` | `std_msgs/String`   | Topic description |

#### Modifications

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Pub/Sub/Srv/Cli | `/old_topic_name` | `sensor_msgs/Image` | Topic description |
| New     | Pub/Sub/Srv/Cli | `/new_topic_name` | `sensor_msgs/Image` | Topic description |

### ROS Parameter Changes

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added/Removed | `param_name`   | `double` | `1.0`         | Param description |

#### Modifications

| Version | Parameter Name   | Type     | Default Value | Description       |
|:--------|:-----------------|:---------|:--------------|:------------------|
| Old     | `old_param_name` | `double` | `1.0`         | Param description |
| New     | `new_param_name` | `double` | `1.0`         | Param description |

🔴⬆️ -->

## Effects on system behavior

None.

kobayu858 commented:
I have renamed all the outer ns.
If you have any other ideas to solve this problem, please let me know.

kobayu858 commented:
Regarding CodeScene Cloud Delta Analysis (main), is it an error that I should ignore because it gives me an error even though I have reduced the number of rows compared to the original function?

veqcc commented on planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/start_planner_module.cpp:
```diff
@@ -628,10 +628,10 @@ BehaviorModuleOutput StartPlannerModule::plan()
     RCLCPP_WARN_THROTTLE(
       getLogger(), *clock_, 5000, "Not found safe pull out path, publish stop path");
-    const auto output = generateStopOutput();
+    const auto stop_output = generateStopOutput();
```
@kobayu858 
Please move the declaration `BehaviorModuleOutput output;` to line 676.
The same applies to `planWaitingApproval` function.

kobayu858 replied:
Thanks for the advice. I have moved it as you suggested to narrow the scope range.

veqcc commented:
> Regarding CodeScene Cloud Delta Analysis (main), is it an error that I should ignore because it gives me an error even though I have reduced the number of rows compared to the original function?

Yes you can ignore it this time :+1:

veqcc commented on planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/manager.cpp:
```diff
@@ -312,8 +303,8 @@ void StartPlannerModuleManager::init(rclcpp::Node * node)
     // ObjectTypesToCheck
-    std::string obj_types_ns = surround_moving_obstacle_check_ns + "object_types_to_check.";
     {
+      std::string obj_types_ns = surround_moving_obstacle_check_ns + "object_types_to_check.";
```
Isn't it a const variable?

kobayu858 replied:
I missed it this time because it was a fix for the shadowVariable warning. I will be careful from now on.

veqcc commented:
> I have renamed all the inner scopes ns.
Also removed extra blank lines to pass CI.
If you have any other ideas to solve this problem, please let me know.

How about declaring all the namespace variables in each code block?
```
{
  const std::string ns = "start_planner.";
  ...
}
{
  const std::string ns = "start_planner.hoge";
  ...
}
{
  const std::string ns = "start_planner.fuga";
  ...
}
```

kobayu858 commented:
Moved all parts using `const std::string ns = "start_planner.";` to one block.

veqcc commented:
The structure of code blocks seem wierd.

If namespaces start with the same prefix, then they should be under a same code block.
For example,

Good
```
{
  // namespace hoge
  {
    // namespace hoge.fuga
  }
  {
    // namespace hoge.piyo
  }
}
```

Bad
```
{
  // namespace hoge
  {
    // namespace hoge.fuga
  }
}
{
  // namespace hoge.piyo
}
```

kobayu858 commented:
Reorganized code blocks to correspond to the original namespace.

kobayu858 commented:
@danielsanchezaran @kosuke55 @kyoichi-sugahara @soblin @satoshi-ota @shmpwk @TomohitoAndo @tkimura4 

Please confirm the PR. Thank you very much.

danielsanchezaran commented:
@kobayu858 have you tested these changes in PSim? what about  parameter updates?
Please confirm that all namespaces are correct in updateModuleParams() by updating at least one parameter per scope using rqt_reconfigure or ros2 param set and checking that the parameter is updated.

kobayu858 commented:
@danielsanchezaran cc: @veqcc 

Thanks for the advice.

First, I ran PSim as the first test.
/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner is up and running fine.
It has been confirmed that it does not start up properly if the namespace is different.

Next, as a second test, I compared the parameters before and after the change.
As a result of the comparison, all 129 parameters were the same, so there seems to be no problem. The command used was `ros2 param list /planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner | grep “start_planner\.” `

Finally, I verified that all namespaces in `updateModuleParams()` were correct by updating at least one parameter per scope using `ros2 param set` and making sure the parameters were updated.
I have also confirmed that if the namespaces are different here, they will not be updated.

kobayu858 commented:
initial state
```
[component_container_mt-32] [INFO] [1721635766.007260284] [StartPlannerModuleManager]: Updated th_arrived_distance: 1.000000
[component_container_mt-32] [INFO] [1721635766.007325593] [StartPlannerModuleManager]: Updated check_car: 1
[component_container_mt-32] [INFO] [1721635766.007337335] [StartPlannerModuleManager]: Updated center_line_path_interval: 1.000000
[component_container_mt-32] [INFO] [1721635766.007350941] [StartPlannerModuleManager]: Updated maximum_deceleration_for_stop: 1.000000
[component_container_mt-32] [INFO] [1721635766.007358209] [StartPlannerModuleManager]: Updated enable_freespace_planner: 1
[component_container_mt-32] [INFO] [1721635766.007367069] [StartPlannerModuleManager]: Updated theta_size: 144
[component_container_mt-32] [INFO] [1721635766.007374984] [StartPlannerModuleManager]: Updated obstacle_threshold: 30
[component_container_mt-32] [INFO] [1721635766.007381678] [StartPlannerModuleManager]: Updated use_back: 0
[component_container_mt-32] [INFO] [1721635766.007388507] [StartPlannerModuleManager]: Updated enable_update: 1
[component_container_mt-32] [INFO] [1721635766.007395642] [StartPlannerModuleManager]: Updated min_velocity: 0.000000
[component_container_mt-32] [INFO] [1721635766.007403871] [StartPlannerModuleManager]: Updated safety_check_time_horizon: 10.000000
[component_container_mt-32] [INFO] [1721635766.007416007] [StartPlannerModuleManager]: Updated check_car: 1
[component_container_mt-32] [INFO] [1721635766.007423761] [StartPlannerModuleManager]: Updated check_current_lane: 1
[component_container_mt-32] [INFO] [1721635766.007430896] [StartPlannerModuleManager]: Updated enable_safety_check: 1
[component_container_mt-32] [INFO] [1721635766.007438340] [StartPlannerModuleManager]: Updated rear_vehicle_reaction_time: 2.000000
[component_container_mt-32] [INFO] [1721635766.007446378] [StartPlannerModuleManager]: Updated search_radius: 10.000000
[component_container_mt-32] [INFO] [1721635766.007454221] [StartPlannerModuleManager]: Updated check_car: 1
[component_container_mt-32] [INFO] [1721635766.007462174] [StartPlannerModuleManager]: Updated print_debug_info: 0
```
Successful
[component_container_mt-32] [INFO] [1721636298.298386631] [StartPlannerModuleManager]: Updated th_arrived_distance: 101.000000
[component_container_mt-32] [INFO] [1721636298.298451034] [StartPlannerModuleManager]: Updated check_car: 0
[component_container_mt-32] [INFO] [1721636298.298467238] [StartPlannerModuleManager]: Updated center_line_path_interval: 101.000000
[component_container_mt-32] [INFO] [1721636298.298484853] [StartPlannerModuleManager]: Updated maximum_deceleration_for_stop: 101.000000
[component_container_mt-32] [INFO] [1721636298.298496278] [StartPlannerModuleManager]: Updated enable_freespace_planner: 0
[component_container_mt-32] [INFO] [1721636298.298508151] [StartPlannerModuleManager]: Updated theta_size: 145
[component_container_mt-32] [INFO] [1721636298.298518817] [StartPlannerModuleManager]: Updated obstacle_threshold: 31
[component_container_mt-32] [INFO] [1721636298.298527862] [StartPlannerModuleManager]: Updated use_back: 1
[component_container_mt-32] [INFO] [1721636298.298536968] [StartPlannerModuleManager]: Updated enable_update: 0
[component_container_mt-32] [INFO] [1721636298.298546671] [StartPlannerModuleManager]: Updated min_velocity: 100.000000
[component_container_mt-32] [INFO] [1721636298.298558925] [StartPlannerModuleManager]: Updated safety_check_time_horizon: 110.000000
[component_container_mt-32] [INFO] [1721636298.298572048] [StartPlannerModuleManager]: Updated check_car: 0
[component_container_mt-32] [INFO] [1721636298.298582210] [StartPlannerModuleManager]: Updated check_current_lane: 0
[component_container_mt-32] [INFO] [1721636298.298591773] [StartPlannerModuleManager]: Updated enable_safety_check: 0
[component_container_mt-32] [INFO] [1721636298.298602074] [StartPlannerModuleManager]: Updated rear_vehicle_reaction_time: 102.000000
[component_container_mt-32] [INFO] [1721636298.298614326] [StartPlannerModuleManager]: Updated search_radius: 110.000000
[component_container_mt-32] [INFO] [1721636298.298625659] [StartPlannerModuleManager]: Updated check_car: 0
[component_container_mt-32] [INFO] [1721636298.298636463] [StartPlannerModuleManager]: Updated print_debug_info: 1
Failed
[component_container_mt-32] [INFO] [1721637124.118711454] [StartPlannerModuleManager]: Updated th_arrived_distance: 1.000000
[component_container_mt-32] [INFO] [1721637124.118785937] [StartPlannerModuleManager]: Updated check_car: 1
[component_container_mt-32] [INFO] [1721637124.118802324] [StartPlannerModuleManager]: Updated center_line_path_interval: 1.000000
[component_container_mt-32] [INFO] [1721637124.118819608] [StartPlannerModuleManager]: Updated maximum_deceleration_for_stop: 1.000000
[component_container_mt-32] [INFO] [1721637124.118831161] [StartPlannerModuleManager]: Updated enable_freespace_planner: 0
[component_container_mt-32] [INFO] [1721637124.118844237] [StartPlannerModuleManager]: Updated theta_size: 145
[component_container_mt-32] [INFO] [1721637124.118855812] [StartPlannerModuleManager]: Updated obstacle_threshold: 31
[component_container_mt-32] [INFO] [1721637124.118865355] [StartPlannerModuleManager]: Updated use_back: 1
[component_container_mt-32] [INFO] [1721637124.118875291] [StartPlannerModuleManager]: Updated enable_update: 0
[component_container_mt-32] [INFO] [1721637124.118884243] [StartPlannerModuleManager]: Updated min_velocity: 100.000000
[component_container_mt-32] [INFO] [1721637124.118896295] [StartPlannerModuleManager]: Updated safety_check_time_horizon: 110.000000
[component_container_mt-32] [INFO] [1721637124.118908069] [StartPlannerModuleManager]: Updated check_car: 0
[component_container_mt-32] [INFO] [1721637124.118917420] [StartPlannerModuleManager]: Updated check_current_lane: 0
[component_container_mt-32] [INFO] [1721637124.118925923] [StartPlannerModuleManager]: Updated enable_safety_check: 0
[component_container_mt-32] [INFO] [1721637124.118934472] [StartPlannerModuleManager]: Updated rear_vehicle_reaction_time: 102.000000
[component_container_mt-32] [INFO] [1721637124.118945630] [StartPlannerModuleManager]: Updated search_radius: 110.000000
[component_container_mt-32] [INFO] [1721637124.118955555] [StartPlannerModuleManager]: Updated check_car: 0
[component_container_mt-32] [INFO] [1721637124.118964495] [StartPlannerModuleManager]: Updated print_debug_info: 1

danielsanchezaran commented:
> @danielsanchezaran cc: @veqcc
> 
> Thanks for the advice.
> 
> First, I ran PSim as the first test. /planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner is up and running fine. It has been confirmed that it does not start up properly if the namespace is different.
> 
> Next, as a second test, I compared the parameters before and after the change. As a result of the comparison, all 129 parameters were the same, so there seems to be no problem. The command used was `ros2 param list /planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner | grep “start_planner\.” `
> 
> Finally, I verified that all namespaces in `updateModuleParams()` were correct by updating at least one parameter per scope using `ros2 param set` and making sure the parameters were updated. I have also confirmed that if the namespaces are different here, they will not be updated.

Thank you very much! LGTM

danielsanchezaran reviewed (approved):
LGTM

kobayu858 commented:
The `end_pose_curvature_threshold` was added and tested again to confirm.
